docs/src/pages/components/slider/CustomizedSlider.js

```diff
@@ -70,80 +71,74 @@ const IOSSlider = withStyles({
       },
     },
   },
-  active: {},
-  valueLabel: {
+  '& .MuiSlider-valueLabel': {
     left: 'calc(-50% + 12px)',
     top: -22,
     '& *': {
       background: 'transparent',
       color: '#000',
     },
   },
-  track: {
+  '& .MuiSlider-track': {
```

Comment:
That's going to be hard to sell.

Reply:
This will only concern v5 users, I think that most developers will prefer to do this than the current `classes` magic. I think that most developers are already used to the CSS selector path, e.g. [Buffet](https://github.com/strapi/buffet/blob/746a0658485b89a6b3744f66dcca86ef183892aa/packages/buffetjs-styles/src/components/DatePicker/index.js#L62),

Reply:
However: https://github.com/mui-org/material-ui/issues/22342#issuecomment-702123299.

Reply:
That's not "however". That's even worse.

Reply:
I don't understand how using global classes is worse than learning some custom logic from a library. I understand that using the `classes` has type safety, but I don't think that is bigger argument, that using CSS in general which has been a standard for a long time. Plus there are ways to add it on top of it if we want to, babel plugin, utility consts etc.

Do you really think this would be a deal breaker?

Reply:
It's going from "what name does this identifier have?" to
1. reference syntax `& ` (and be careful with the spacing here)
1. where to import the classname from
1. reference the classname with template literals

There's also just so much syntax going on now. 

I don't think we'll get anywhere by discussing this. Ideally it should've been part of the RFC process. This is why I generally don't care about upvotes on issues. It's just way too inaccurate to begin with and even more inaccurate if the issue isn't accurate. Saying "switch to emotion" is easy to sell if you don't include the concrete API/syntax you're migrating to. But we can only look forward at this point and hope that it pays off.

Reply:
> It's going from "what name does this identifier have?" to

@eps1lon It will likely be simpler for most developers in v5. We have been regularly linking https://material-ui.com/styles/advanced/#with-material-ui-core in v4 when developers were struggling to understand how this `classes` API work. It's not obvious for many. I think that the new syntax will work better because this it's closer to what developers are already familiar with: CSS/SASS. Another positive signal is the volume of complaints around hard to customize components seems to have gone down between v3 and v4 once we introduced the global class names.

> Ideally it should've been part of the RFC process.

This is one of the first dimensions we have considered. I believe the section "Deterministic classnames on the components that can be targeted for custom styles" surfaces this, but we could have exposed it deeper.